internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"DISABLED", "ENABLED", "REQUIRED"}...),
+				},
+			},
+			"client_secret_retention_period": schema.Int64Attribute{
+				Description: "The length of time in minutes that client secrets will be retained as secondary secrets after secret change. The default value is 0, which will disable secondary client secret retention.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"jwt_secured_authorization_response_mode_lifetime": schema.Int64Attribute{
+				Description: "The lifetime, in seconds, of the JWT Secured authorization response. The default value is 600.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+
+	// Set attributes in string list
+	if setOptionalToComputed {
+		config.SetAllAttributesToOptionalAndComputed(&schema, []string{"default_scope_description", "authorization_code_timeout", "authorization_code_entropy", "refresh_token_length", "refresh_rolling_interval", "registered_authorization_path", "pending_authorization_timeout", "device_polling_interval", "bypass_activation_code_confirmation"})
+	}
+	config.AddCommonSchema(&schema, false)
+	resp.Schema = schema
+}
+
+func (r *oauthAuthServerSettingsResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+
+	var model oauthAuthServerSettingsResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	// Validate allowed_origins value(s)
+	if internaltypes.IsDefined(model.AllowedOrigins) {
+		aoElems := model.AllowedOrigins.Elements()
+		for _, aoElem := range aoElems {
+			aoElemString := aoElem.(basetypes.StringValue).ValueString()
+			isElemUrl := internaltypes.IsUrlFormat(aoElemString)
+			if !isElemUrl {
+				resp.Diagnostics.AddError("Invalid URL Format!", fmt.Sprintf("Please provide a valid origin. Origin \"%s\" needs to be in a valid URL-like format - \"http(s)//:<value>.<domain>\"", aoElemString))
+			}
+		}
+	}
+
+	// Scope list for comparing values in matchNameBtwnScopes variable
+	scopeNames := []string{}
+
+	// Test scope names for dynamic true, string must be prepended with *
+	if internaltypes.IsDefined(model.Scopes) {
+		scopeElems := model.Scopes.Elements()
+		for _, scopeElem := range scopeElems {
+			scopeElemObjectAttrs := scopeElem.(types.Object)
+			scopeEntryName := scopeElemObjectAttrs.Attributes()["name"].(basetypes.StringValue).ValueString()
+			scopeNames = append(scopeNames, scopeEntryName)
+			scopeEntryIsDynamic := scopeElemObjectAttrs.Attributes()["dynamic"].(basetypes.BoolValue).ValueBool()
+			if scopeEntryIsDynamic {
+				if string(scopeEntryName[0]) != "*" {
```

Comment:
Add a check for string length >= 1 here to avoid a panic.

Also I'm not sure if this is a safe way to get the first character (rune) of a string in go - it might not handle non-ASCII characters well. We could write some utility function that converts a string to a slice of runes potentially.